strongbox-storage/strongbox-storage-api/src/main/java/org/carlspring/strongbox/storage/snapshot/MavenSnapshotManager.java

```diff
@@ -0,0 +1,265 @@
+package org.carlspring.strongbox.storage.snapshot;
+
+import org.carlspring.maven.commons.util.ArtifactUtils;
+import org.carlspring.strongbox.providers.ProviderImplementationException;
+import org.carlspring.strongbox.providers.layout.LayoutProvider;
+import org.carlspring.strongbox.providers.layout.LayoutProviderRegistry;
+import org.carlspring.strongbox.providers.storage.StorageProviderRegistry;
+import org.carlspring.strongbox.storage.metadata.MavenMetadataManager;
+import org.carlspring.strongbox.storage.metadata.VersionCollectionRequest;
+import org.carlspring.strongbox.storage.repository.Repository;
+
+import javax.inject.Inject;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.security.NoSuchAlgorithmException;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import edu.emory.mathcs.backport.java.util.Arrays;
+import org.apache.maven.artifact.Artifact;
+import org.apache.maven.artifact.repository.metadata.Metadata;
+import org.apache.maven.artifact.repository.metadata.Versioning;
+import org.codehaus.plexus.util.xml.pull.XmlPullParserException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.stereotype.Component;
+import static org.carlspring.strongbox.providers.layout.LayoutProviderRegistry.getLayoutProvider;
+
+/**
+ * @author Kate Novik.
+ */
+@Component
+public class MavenSnapshotManager
+{
+
+    private static final String TIMESTAMP_FORMAT = "yyyyMMdd.HHmmss";
+
+    private static final Logger logger = LoggerFactory.getLogger(MavenSnapshotManager.class);
+
+    @Inject
+    private LayoutProviderRegistry layoutProviderRegistry;
+    @Inject
+    protected StorageProviderRegistry storageProviderRegistry;
+    @Inject
+    private MavenMetadataManager mavenMetadataManager;
+
+
+    public MavenSnapshotManager()
+    {
+    }
+
+    public void deleteTimestampedSnapshotArtifacts(Repository repository,
+                                                   String artifactPath,
+                                                   VersionCollectionRequest request,
+                                                   int numberToKeep,
+                                                   int keepPeriod)
+            throws IOException, ProviderImplementationException,
```

Comment:
Could you please put all exceptions on a separate line, and do the same for all multi-line method parameters across the pull? Thanks! :)